### PR TITLE
Use go1.21

### PR DESCRIPTION
**Pull request by brianstrauch**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Use new built-in functions and `slices` package. Plus, a speed boost in build times!

Test & Review
-------------
Tests still pass

### CODE REVIEW

**Comment by brianstrauch** [comment]
(Looks like go1.21 isn't supported by Semaphore yet)

**Review by sgagniere — APPROVED** [review]
Very nice, it looks much cleaner now!

Is the lack of 1.21 support in Semaphore a blocker for merging this PR?

**Comment by brianstrauch** [comment]
> Is the lack of 1.21 support in Semaphore a blocker for merging this PR?

Yeah, usually it takes a few weeks for them to add it, IIRC

File: pkg/types/slice.go [hunk 1]
```diff
@@ -17,8 +17,3 @@ func Remove(haystack []string, needle string) []string {
 	}
```
Comment (kevin-wu24) [inline pkg/types/slice.go:17]: `slices` has a `Contains` and `DeleteFunc` functions, so we can remove the `Contains` and `Remove` functions above.

File: pkg/utils/utils_test.go [hunk 2]
```diff
@@ -19,12 +18,12 @@ const (
...
 func TestContains(t *testing.T) {
 	req := require.New(t)
-	req.True(types.Contains([]string{"a"}, "a"))
+	req.True(slices.Contains([]string{"a"}, "a"))
 }
 
 func TestDoesNotContain(t *testing.T) {
 	req := require.New(t)
-	req.False(types.Contains([]string{}, "a"))
+	req.False(slices.Contains([]string{}, "a"))
```
Comment (kevin-wu24) [inline pkg/utils/utils_test.go:26]: I don't think we need these tests anymore.

File: pkg/panic-recovery/panic_recovery.go [hunk 3]
```diff
@@ -3,11 +3,11 @@ package panic_recovery
 import (
 	"runtime"
 	"runtime/debug"
+	"slices"
 	"strings"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
-	"golang.org/x/exp/slices"
```
Comment (kevin-wu24) [inline pkg/panic-recovery/panic_recovery.go:10]: The `configuration/command.go` that I merged today also import this, so if you could replace that too that'd be nice!

File: pkg/utils/utils.go [hunk 4]
```diff
@@ -13,13 +13,6 @@ import (
...
-func Max(x, y int64) int64 {
-	if x > y {
-		return x
-	}
-	return y
-}
-
 func TestEq(a, b []string) bool {
 	// If one is nil, the other must also be nil.
 	if (a == nil) != (b == nil) {
```
Comment (kevin-wu24) [inline pkg/utils/utils.go:18]: `slices` has a `Equal` function now, so we can remove this.

File: pkg/types/slice.go (hunk 1 above)
Comment (brianstrauch) [reply pkg/types/slice.go]: Thanks, I deleted these but they got re-added in a bad merge 🤦‍♂️

File: pkg/utils/utils.go (hunk 4 above)
Comment (brianstrauch) [reply pkg/utils/utils.go:18]: Nice find!

**Review by kevin-wu24 — APPROVED** [review]

**Pull request merged**